pst/actions/transfer.ts

```diff
@@ -0,0 +1,29 @@
+import {ActionSchema, HandlerResultSchema, StateSchema} from "../schemas";
+import {Transaction} from "../imports/smartweave/transaction";
+
+export function transfer(state: StateSchema, action: ActionSchema): HandlerResultSchema {
+    const target = action.target;
+    const qty = action.qty;
+    const caller = Transaction.owner();
+
+    if (qty <= 0 || caller === target) {
+        throw new Error('[CE:ITT] Invalid token transfer');
+    }
+
+    if (state.balances.has(caller) && state.balances.get(caller) < qty) {
```

Comment:
hmm, shouldn't it rather be
`(!state.balances.has(caller) || state.balances.get(caller) < qty)` ?